### DISCUSSION TITLE
Disabling translations of slugs

### DISCUSSION THREAD

**Discussion by NicTorgersen** [spatie/laravel-tags#317] [Q&A]
I don't care for translating my slugs as they are largely programmatic components. Slugs are used in our search-repository to provide a faceted search experience.

I tried extending `Spatie\Tags\Tag` and overriding the `$translatable` array, but to no real avail as it looks like the `HasSlugs`-trait expects slugs to be translatable.

Has anyone been in the same boat who can provide some thoughts or tips?

**Comment by danielschweiger** [comment]
I had a similar requirement, I wanted to use tags only in German. Even when current locale is not "de", only "de" tags should be used (fetched or created).

First I did this:
https://spatie.be/docs/laravel-tags/v4/advanced-usage/using-another-default-locale

I removed this function:

```
public static function getLocale(): string
    {
        return 'en';
    }
```

and used it this way:

```
public static function getLocale(): string
    {
        return 'de';
    }
```

Then I overwrote this function im my class:

```
public static function findOrCreate($values, string $type = null, string $locale = null)
    {
        $locale = 'de';

        $tags = collect($values)->map(function ($value) use ($type, $locale) {
            if ($value instanceof self) {
                return $value;
            }

            return static::findOrCreateFromString($value, $type, $locale);
        });

        return is_string($values) ? $tags->first() : $tags;
    }
```

There I set $locale to "de".
